AsyncDisplayKit/ASTextNode.mm

```diff
@@ -48,30 +42,16 @@ - (instancetype)initWithRenderer:(ASTextKitRenderer *)renderer
 
 @implementation ASTextNodeDrawParameters
 
-- (instancetype)initWithRenderer:(ASTextKitRenderer *)renderer
-                      textOrigin:(CGPoint)textOrigin
-                 backgroundColor:(UIColor *)backgroundColor
+- (instancetype)initWithBounds:(CGRect)bounds
+               backgroundColor:(UIColor *)backgroundColor
 {
   if (self = [super init]) {
-    _renderer = renderer;
-    _textOrigin = textOrigin;
+    _bounds = bounds;
     _backgroundColor = backgroundColor;
   }
   return self;
 }
 
-- (void)dealloc
```

Comment:
Unless I'm missing something, please re add this. I recently audited the necessity of this code, and it is currently very important, although someday I would like to be able to make sure that system provided nodes--where we can ensure their thread safety on dealloc--can be de-allocated off the main thread. This is what the Paper app is able to do with its nodes. The reason this is not the case in the open source framework is that we cannot guarantee that a node will not have an instance variable for something like a gesture recognizer which must not be released off the main thread.